src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/SendOrderContestedAboutToStartHandler.java

```diff
@@ -0,0 +1,140 @@
+        log.info("Event {} fetching all partys solicitor case role for caseId {}", EventType.SEND_ORDER, caseDetails.getId());
+
+        FinremCaseData caseData = caseDetails.getData();
+        List<DynamicMultiSelectListElement> dynamicListElements = new ArrayList<>();
+        List<DynamicMultiSelectListElement> defaultDynamicListElements = new ArrayList<>();
+
+        if (caseData.getApplicantOrganisationPolicy() != null && caseData.getApplicantOrganisationPolicy().getOrgPolicyCaseAssignedRole() != null) {
+            String assignedAppRole = caseData.getApplicantOrganisationPolicy().getOrgPolicyCaseAssignedRole();
+            DynamicMultiSelectListElement appMultiSelectListElement = generalOrderService.getDynamicMultiSelectListElement(assignedAppRole,
+                DISPLAY_LABEL.formatted(APPLICANT, caseData.getFullApplicantName()));
+            dynamicListElements.add(appMultiSelectListElement);
+            defaultDynamicListElements.add(appMultiSelectListElement);
+        }
+
+        if (caseData.getRespondentOrganisationPolicy() != null && caseData.getRespondentOrganisationPolicy().getOrgPolicyCaseAssignedRole() != null) {
+            String assignedRespRole = caseData.getRespondentOrganisationPolicy().getOrgPolicyCaseAssignedRole();
+            DynamicMultiSelectListElement respMultiSelectListElement = generalOrderService.getDynamicMultiSelectListElement(assignedRespRole,
+                DISPLAY_LABEL.formatted(RESPONDENT, caseData.getRespondentFullName()));
+            dynamicListElements.add(respMultiSelectListElement);
+            defaultDynamicListElements.add(respMultiSelectListElement);
+        }
+        return getRoleList(intervenerCaseRoleList(caseData, dynamicListElements),
+            caseDetails.getData().getPartiesOnCase(), defaultDynamicListElements);
+    }
+
+    private DynamicMultiSelectList getRoleList(List<DynamicMultiSelectListElement> dynamicMultiSelectListElement,
+                                               DynamicMultiSelectList selectedRoles,
+                                               List<DynamicMultiSelectListElement> defaultDynamicListElements) {
+        if (selectedRoles != null) {
+            return DynamicMultiSelectList.builder()
+                .value(selectedRoles.getValue())
+                .listItems(dynamicMultiSelectListElement)
+                .build();
+        } else {
+            return DynamicMultiSelectList.builder()
+                .value(defaultDynamicListElements)
+                .listItems(dynamicMultiSelectListElement)
+                .build();
+        }
+    }
+
+    private List<DynamicMultiSelectListElement> intervenerCaseRoleList(FinremCaseData caseData,
+                                                                       List<DynamicMultiSelectListElement> dynamicListElements) {
+        //intervener1
+        IntervenerWrapper oneWrapper = caseData.getIntervenerOneWrapperIfPopulated();
```

Comment:
Refactor this chunk of code into a reusable to set dynamic list for intervener to avoid duplication for all other interveners

Reply:
moved this into PartyService